src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -208,6 +218,16 @@ interval_sparse_arrayt::of_expr(const exprt &expr, const exprt &extra_value)
   return {};
 }
 
+exprt interval_sparse_arrayt::at(std::size_t index) const
+{
+  for(const auto &pair : entries)
+  {
+    if(pair.first >= index)
+      return pair.second;
```

Comment:
If it's worth it, you could rewrite `entries` to as `std::map` and then use this function to do this in `O(log n)`:
http://www.cplusplus.com/reference/map/map/upper_bound/